packages/d2mini/tests/multiset.test.ts

```diff
@@ -108,3 +108,101 @@ describe('MultiSet', () => {
     ])
   })
 })
+
+describe('LazyMultiSet', () => {
+  describe('basic operations', () => {
+    let a: LazyMultiSet<[string, string | string[]]>
+    let b: LazyMultiSet<[string, string | string[]]>
+
+    beforeEach(() => {
+      a = LazyMultiSet.fromArray([
+        [['apple', '$5'], 2],
+        [['banana', '$2'], 1],
+      ])
+      b = LazyMultiSet.fromArray([
+        [['apple', '$3'], 1],
+        [['apple', ['granny smith', '$2']], 1],
+        [['kiwi', '$2'], 1],
+      ])
+    })
+
+    it('should concatenate two lazy multisets', () => {
+      const concat = a.concat(b)
+      expect(concat.getInner()).toEqual([
+        [['apple', '$5'], 2],
+        [['banana', '$2'], 1],
+        [['apple', '$3'], 1],
+        [['apple', ['granny smith', '$2']], 1],
+        [['kiwi', '$2'], 1],
+      ])
+    })
+
+    it('should filter elements lazily', () => {
+      const filtered = a.filter((data) => data[0] !== 'apple')
+      expect(filtered.getInner()).toEqual([[['banana', '$2'], 1]])
+    })
+
+    it('should map elements lazily', () => {
+      const mapped = a.map((data) => [data[1], data[0]])
+      expect(mapped.getInner()).toEqual([
+        [['$5', 'apple'], 2],
+        [['$2', 'banana'], 1],
+      ])
+    })
+
+    it('should be iterable', () => {
+      const result: [[string, string | string[]], number][] = []
+      for (const entry of a) {
+        result.push(entry)
+      }
+      expect(result).toEqual([
+        [['apple', '$5'], 2],
+        [['banana', '$2'], 1],
+      ])
+    })
+
+    it('should negate multiplicities', () => {
+      const negated = a.negate()
+      expect(negated.getInner()).toEqual([
+        [['apple', '$5'], -2],
+        [['banana', '$2'], -1],
+      ])
+    })
+  })
+
+  it('should consolidate correctly', () => {
```

Comment:
Let's also check the corner case where the multiplicities for a certain value cancel out and that the result does not contain the value.